webapp/src/components/common/multiSelect/multiselect.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import type {GroupBase, MultiValue, SingleValue} from 'react-select';
+import Select from 'react-select';
+import type {SelectComponentsConfig} from 'react-select/dist/declarations/src/components';
+
+import type {DropdownOption} from 'components/common/dropdown/dropdown';
+export type CustomComponentsDefinition = SelectComponentsConfig<DropdownOption, true, GroupBase<DropdownOption>>
+
+export type Props = {
+    options: DropdownOption[];
+    values?: DropdownOption[];
+    customComponents?: CustomComponentsDefinition;
+    onChange: (selectedValues: DropdownOption[]) => void;
+}
+
+function Multiselect({options, values, customComponents, onChange}: Props) {
+    const onChangeHandler = useCallback((newValue: SingleValue<DropdownOption> | MultiValue<DropdownOption>) => {
+        if (Array.isArray(newValue)) {
+            onChange(newValue);
+        }
```

Comment:
This is a multiselect component, so the `newValue` will always be an array. However the underlying react-select component supports both single and multi select, so  its change handler wither receives a `DropdownOption`, or an array of `DropdownOption`. Since I know this is a multiselect component, I know it will always be an array, so I check for array as a way to satisfy type checks. The change handler of this custom component always expects an array, so either I can force `newValue as DropdownOption[]`, or check it this way.